src/azure-cli/azure/cli/command_modules/servicefabric/_arm_deployment_utils.py

```diff
@@ -0,0 +1,63 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import datetime
+
+from azure.cli.core.util import CLIError, sdk_no_wait
+from azure.cli.core.commands import LongRunningOperation
+from knack.log import get_logger
+from ._client_factory import (resource_client_factory)
+
+logger = get_logger(__name__)
+
+
+def validate_and_deploy_arm_template(cli_ctx, resource_group_name, template, parameters):
+    suffix = datetime.datetime.now().strftime("%Y%m%d%H%M")
+    deployment_name = 'AzurePSDeployment-' + suffix
+
+    logger.info("Validating the deployment")
+    validate_result = _deploy_arm_template_core(
+        cli_ctx, resource_group_name, template, parameters, deployment_name, 'incremental', True)
+    if validate_result.error is not None:
+        errors_detailed = _build_detailed_error(validate_result.error, [])
+        errors_detailed.insert(0, "Error validating template. See below for more information.")
+        raise CLIError('\n'.join(errors_detailed))
+    logger.info("Deployment is valid, and begin to deploy")
+    _deploy_arm_template_core(cli_ctx, resource_group_name, template,
+                              parameters, deployment_name, 'incremental', False)
+
+
+def _deploy_arm_template_core(cli_ctx,
+                              resource_group_name,
+                              template,
+                              parameters,
+                              deployment_name=None,
+                              mode='incremental',
+                              validate_only=False,
+                              no_wait=False):
+    from azure.mgmt.resource.resources.models import DeploymentProperties
```

Comment:
Becaues resources module has api constraint, please refer to https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/network/custom.py#L148

Reply:
fixed